src/utils/getStyleWithPreferredLanguage.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+const recurseExpression = (exp: any, prevPropertyRegex: RegExp, nextProperty: string): any => {
+	if (!Array.isArray(exp)) return exp;
+	if (exp[0] !== "coalesce") return exp.map(v => recurseExpression(v, prevPropertyRegex, nextProperty));
+
+	const first = exp[1];
+	const second = exp[2];
+
+	let isMatch = Array.isArray(first) && first[0] === "get" && !!first[1].match(prevPropertyRegex)?.[0];
+
+	isMatch = isMatch && Array.isArray(second) && second[0] === "get";
+	isMatch = isMatch && !exp?.[4];
+
+	if (!isMatch) return exp.map(v => recurseExpression(v, prevPropertyRegex, nextProperty));
+
+	return ["coalesce", ["get", nextProperty], ["get", "name:en"], ["get", "name"]];
+};
+
+const updateLayer = (layer: any, prevPropertyRegex: RegExp, nextProperty: string) => {
+	const nextLayer = {
+		...layer,
+		layout: {
+			...layer.layout,
+			"text-field": recurseExpression(layer.layout["text-field"], prevPropertyRegex, nextProperty)
+		}
+	};
+	return nextLayer;
+};
+
+const setPreferredLanguage = (style: any, language: string) => {
+	const nextStyle = { ...style };
+
+	nextStyle.layers = nextStyle.layers.map((l: { type: string; layout: { [x: string]: any } }) => {
+		if (l.type !== "symbol" || !l?.layout?.["text-field"]) return l;
+		return updateLayer(l, /^name:([A-Za-z\-\_]+)$/g, `name:${language}`);
+	});
+
+	return nextStyle;
+};
+
+export const getStyleWithPreferredLanguage = async (styleUrl: string, preferredLanguage: string) => {
```

Comment:
Not as part of this PR, but we should identify a common place to put this utility so that other customers can leverage this.